src/services/machineLearning/imageSceneService.ts

```diff
@@ -0,0 +1,84 @@
+import * as tf from '@tensorflow/tfjs-core';
+import * as tfjsConverter from '@tensorflow/tfjs-converter';
+import {
+    ObjectDetection,
+    SceneDetectionMethod,
+    SceneDetectionService,
+    Versioned,
+} from 'types/machineLearning';
+import sceneMap from 'utils/machineLearning/sceneMap';
+
+class ImageScene implements SceneDetectionService {
+    method: Versioned<SceneDetectionMethod>;
+    model: tfjsConverter.GraphModel;
+
+    public constructor() {
+        this.method = {
+            value: 'Image-Scene',
+            version: 1,
+        };
+    }
+
+    private async init() {
+        if (this.model) {
+            return;
+        }
+
+        const model = await tfjsConverter.loadGraphModel(
+            '/models/imagescene/model.json'
+        );
+        console.log('loaded image-scene model', model, tf.getBackend());
+        this.model = model;
+    }
+
+    async detectScenes(image: ImageBitmap, minScore: number) {
+        await tf.ready();
+
+        if (!this.model) {
+            await this.init();
+        }
+
+        const output = tf.tidy(() => {
```

Comment:
we should to clean up the tensor with tf.dispose(result) or result.dispose(); after we are done with